### PR TITLE
docs: Update docs for status conditions

### PR DESCRIPTION
**Pull request by jigisha620**
Fixes #N/A 

**Description**
This PR updates docs to add information about status conditions.

**How was this change tested?**
NA

**Does this change impact docs?**
- [x] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/v1.0/concepts/nodeclasses.md [hunk 1]
```diff
@@ -1525,28 +1525,14 @@ status:
...
-  role: "KarpenterNodeRole-${CLUSTER_NAME}"
-status:
-  conditions:
-    Last Transition Time:  2024-05-06T06:19:46Z
-    Message:               unable to resolve instance profile for node class
-    Reason:                NodeClassNotReady
-    Status:                False
-    Type:                  Ready
-```
+If a NodeClass is not ready, it will not be considered for scheduling.
```
Comment (jonathan-innis) [inline website/content/en/v1.0/concepts/nodeclasses.md:1538]:
```suggestion
If a NodeClass is not ready, NodePools that reference it through their `nodeClassRef` will not be considered for scheduling.
```

File: website/content/en/v1.0/concepts/nodeclasses.md [hunk 2]
```diff
@@ -1525,28 +1525,14 @@ status:
...
-    Last Transition Time:  2024-05-06T06:04:45Z
-    Message:               Ready
-    Reason:                Ready
-    Status:                True
-    Type:                  Ready
-```
+NodeClass will have the following status conditions -
 
-If any of the underlying conditions are not resolved then `Status` is `False` and `Message` indicates the dependency that was not resolved.
+| Status Condition     | Description                                                                                                                                                                                                                       |
```
Comment (jonathan-innis) [inline website/content/en/v1.0/concepts/nodeclasses.md:1530]:
```suggestion
| Condition Type     | Description                                                                                                                                                                                                                       |
```

File: website/content/en/v1.0/concepts/nodepools.md [hunk 3]
```diff
@@ -417,6 +417,16 @@ For more information on weighting NodePools, see the [Weighted NodePools section
 * The `status.conditions.reason` object indicates the reason for the condition's previous transition.
 * The `status.conditions.message` object provides human-readable details about the condition's previous transition.
 
+NodePool will have the following status conditions -
+
+| Status Condition    | Description                                                                                                                                       |
```
Comment (jonathan-innis) [inline website/content/en/v1.0/concepts/nodepools.md:422]:
```suggestion
| Condition Type    | Description                                                                                                                                       |
```

File: website/content/en/v1.0/concepts/nodepools.md [hunk 4]
```diff
@@ -417,6 +417,16 @@ For more information on weighting NodePools, see the [Weighted NodePools section
 * The `status.conditions.reason` object indicates the reason for the condition's previous transition.
 * The `status.conditions.message` object provides human-readable details about the condition's previous transition.
 
+NodePool will have the following status conditions -
```
Comment (jonathan-innis) [inline website/content/en/v1.0/concepts/nodepools.md:420]:
```suggestion
NodePools have the following status conditions:
```

File: website/content/en/v1.0/concepts/nodeclasses.md [hunk 5]
```diff
@@ -1525,28 +1525,14 @@ status:
...
-  role: "KarpenterNodeRole-${CLUSTER_NAME}"
-status:
-  conditions:
-    Last Transition Time:  2024-05-06T06:04:45Z
-    Message:               Ready
-    Reason:                Ready
-    Status:                True
-    Type:                  Ready
-```
+NodeClass will have the following status conditions -
```
Comment (jonathan-innis) [inline website/content/en/v1.0/concepts/nodeclasses.md:1528]:
```suggestion
NodeClasses have the following status conditions:
```

File: website/content/en/v1.0/upgrading/v1-migration.md [hunk 6]
```diff
@@ -353,6 +353,8 @@ Karpenter should now be pulling and operating against the v1beta1 APIVersion as
   * Karpenter now adds a `karpenter.sh/unregistered:NoExecute` taint to nodes in injected UserData when using alias in AMISelectorTerms or non-Custom AMIFamily. When using `amiFamily: Custom`, users will need to add this taint into their UserData, where Karpenter will automatically remove it when provisioning nodes.
   * Discovered standard AL2023 AMIs will no longer be considered compatible with GPU / accelerator workloads. If you're using an AL2023 EC2NodeClass (without AMISelectorTerms) for these workloads, you will need to select your AMI via AMISelectorTerms (non-alias).
   * Karpenter now waits for underlying instances to be completely terminated before removing the associated nodes. This means it may take longer for nodes to be deleted and for nodeclaims to get cleaned up.
+  * NodePool now has [status conditions]({{< relref "../concepts/nodepools/#statusconditions" >}}) that indicate if it is ready. If not, then it will not be considered during provisioning.
```
Comment (jonathan-innis) [inline website/content/en/v1.0/upgrading/v1-migration.md:356]:
```suggestion
  * NodePools now have [status conditions]({{< relref "../concepts/nodepools/#statusconditions" >}}) that indicate if they are ready. If not, then they will not be considered during scheduling.
```

File: website/content/en/v1.0/upgrading/v1-migration.md [hunk 7]
```diff
@@ -353,6 +353,8 @@ Karpenter should now be pulling and operating against the v1beta1 APIVersion as
...
+  * NodeClass now has [status conditions]({{< relref "../concepts/nodeclasses/#statusconditions" >}}) that indicate if it is ready. If not, then it will not be considered during provisioning.
```
Comment (jonathan-innis) [inline website/content/en/v1.0/upgrading/v1-migration.md:357]:
```suggestion
  * NodeClasses now has [status conditions]({{< relref "../concepts/nodeclasses/#statusconditions" >}}) that indicate if they are ready. If they are not ready, NodePools that reference them through their `nodeClassRef` will not be considered during scheduling.
```

**Review by jonathan-innis — COMMENTED** [review]
We need to add this to v1.0, preview, and docs so that we keep carrying over the changes that you made here

**Comment by jigisha620** [comment]
> We need to add this to v1.0, preview, and docs so that we keep carrying over the changes that you made here

Agreed, I just wanted to get approval on v1 changes so then I can copy them over.

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**